src/Microsoft.PowerShell.Commands.Utility/commands/utility/CsvCommands.cs

```diff
@@ -989,11 +1014,26 @@ internal string ConvertPSObjectToCSV(PSObject mshObject, IList<string> propertyN
                     _outputString.Append(_delimiter);
                 }
 
-                // If property is not present, assume value is null and skip it.
-                if (mshObject.Properties[propertyName] is PSPropertyInfo property)
+                string value = null;
+                if (mshObject.BaseObject is IDictionary dictionary)
                 {
-                    var value = GetToStringValueForProperty(property);
+                    if (dictionary.Contains(propertyName))
+                    {
+                        value = dictionary[propertyName].ToString();
+                    }
+                    else if (mshObject.Properties[propertyName] is PSPropertyInfo property)
+                    {
+                        value = GetToStringValueForProperty(property);
+                    }
+                }
+                else if (mshObject.Properties[propertyName] is PSPropertyInfo property)
+                {
+                    value = GetToStringValueForProperty(property);
+                }
 
+                // If value is null, assume property is not present and skip it.
+                if (value != null)
```

Comment:
It is not right. We should put in _outputString the value for every property if it should be quoted.
I think we should remove the check.

Reply:
This behaviour is not different to the original behaviour, it is simply consolidating the check so that we don't have to re-check the existence of the property again. 

Is the original behaviour incorrect? How can we test for this? 🙂

Reply:
Sorry. I thought I changed this previously. I expect that for QuoteKind.Always we always put "" in the case.  It is not for the PR although.